dotcom-rendering/src/lib/discussion.ts

```diff
@@ -117,67 +118,38 @@ const comment: BaseSchema<CommentType> = object({
 	),
 });
 
-const discussionApiCommentSuccessSchema = variant('status', [
+export type TopLevelCommentType = Output<typeof topLevelCommentSchema>;
+
+const topLevelCommentSchema = merge([
+	comment,
+	object({
+		responses: optional(array(comment), []),
+	}),
+]);
+
+const commentRepliesResponseSchema = variant('status', [
 	object({
 		status: literal('error'),
 	}),
 	object({
 		status: literal('ok'),
-		comment,
+		comment: topLevelCommentSchema,
 	}),
 ]);
 
 export const parseCommentRepliesResponse = (
 	data: unknown,
 ): Result<'ParsingError' | 'ApiError', CommentType[]> => {
-	const result = safeParse(discussionApiCommentSuccessSchema, data);
+	const result = safeParse(commentRepliesResponseSchema, data);
 	if (!result.success) {
 		return error('ParsingError');
 	}
 	if (result.output.status === 'error') {
 		return error('ApiError');
 	}
-	return ok(result.output.comment.responses ?? []);
+	return ok(result.output.comment.responses);
 };
 
-export interface CommentType {
```

Comment:
This type is now derived from the `comment` schema which is no longer recursive.